### ISSUE TITLE
Make it available on Windows

### ISSUE THREAD

**Issue by yasharya2901** [Rafostar/clapper#103]
I know this is an exclusive app for linux but making it available for other platforms can help in widespread adoption and will definitely help in competing with VLC, even though that's not what you want, it will ensure the long time survival of the project you created.

**Comment by Rafostar** [comment]
I am not competing with VLC. The only thing I do is make it how I want it and share it. Right now the only linux shells that this project targets are GNOME and Phosh (hence I could use JS to assemble the UI). Other linux desktops can just install it as a flatpak.

Even if ported to said platforms (or even used on a non-GNOME distros) the UI would not look right. It uses GTK as all GNOME apps. It would also be very hard to maintain the compatibility on all these platforms by a single person. Not to mention that we would need Vulkan video decoding in GStreamer and an efficient way of drawing them in GTK4. Unlike OpenGL neither one of those is available at the moment.

**Comment by maxloh** [comment]
> I am not competing with VLC. The only thing I do is make it how I want it and share it. Right now the only linux shells that this project targets are GNOME and Phosh (hence I could use JS to assemble the UI). Other linux desktops can just install it as a flatpak.
> 
> Even if ported to said platforms (or even used on a non-GNOME distros) the UI would not look right. It uses GTK as all GNOME apps. It would also be very hard to maintain the compatibility on all these platforms by a single person. Not to mention that we would need Vulkan video decoding in GStreamer and an efficient way of drawing such buffers in GTK4. Unlike OpenGL neither one of those is available at the moment.

It seems that GTK supports Windows.
https://www.gtk.org/docs/installations/windows

**Comment by Rafostar** [comment]
Once new video sink lands and we port away from GJS, this might actually work.

**Comment by Rafostar** [comment]
Its not yet possible to run app as a whole outside of Linux, but the recently merged new sink can be compiled and tried on Windows and probably on macOS too (haven't tried the latter one as I do not have a Mac).

For anyone interested in building sink alone:
```sh
meson builddir -Dplayer=false -Dlib=false -Dgst-plugin=enabled -Dgluploader=enabled
cd builddir
meson install
```
And to test:
```sh
 GST_PLUGIN_FEATURE_RANK=d3d11h264dec:300 gst-play-1.0 --videosink=clappersink 'video.mp4'
```
Eventually replace `d3d11h264dec` with `nvh264dec` if you are using Nvidia GPU to test with NVDEC.

Screenshot of working playback:
![clappersink-win2](https://user-images.githubusercontent.com/40623528/170479675-5fdac4dc-dbed-4838-9710-252422f1b61c.PNG)

**Comment by Rafostar** [comment]
Lets limit this issue here to Windows support. Chances are it already builds on MacOS, but I do not have an Apple computer to test, so if it doesn't please open a separate issue for MacOS.

**Comment by Rafostar** [comment]
Now Clapper works on MS Windows. Builds are available for download [here](https://github.com/Rafostar/clapper/actions/workflows/windows.yml).
